app/src/main/java/com/example/capsule/ui/outfits/OutfitsFragment.kt

```diff
@@ -32,9 +48,48 @@ class OutfitsFragment : Fragment() {
         outfitsViewModel.text.observe(viewLifecycleOwner) {
             textView.text = it
         }
+
+        checkPermission()
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(requireActivity())
+        println("DEBUG season: " + Util.determineSeason(temp))
         return root
     }
 
+    private fun getLastLocation(){
+        checkPermission()
+        val locationTask : Task<Location> = fusedLocationProviderClient.lastLocation
+
+        locationTask.addOnSuccessListener { OnSuccessListener<Location> {
+            location = it
+            getWeather()
+        }}
+    }
+
+    private fun getWeather(){
+        weatherApi = WeatherApi()
+        weatherApi.getWeather(location)
+        temp = weatherApi.getTemp()
+        precip = weatherApi.getPrecip()
+    }
+
+
+    private fun checkPermission() {
+        if (ContextCompat.checkSelfPermission(requireActivity(), Manifest.permission.ACCESS_FINE_LOCATION) !=
+            PackageManager.PERMISSION_GRANTED) ActivityCompat.requestPermissions(requireActivity(), arrayOf(
+            Manifest.permission.ACCESS_FINE_LOCATION), 0)
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
```

Comment:
This function is never called for me and so the temp value is never changing; it's always 0.0 so the season is always winter for me

Reply:
revised and changed to a different function